doc/TOR.md

```diff
@@ -1,144 +1,302 @@
+
+You will be reachable via Clearnet, via TOR to the .onion V3 address and the .onion V2 address if this
+last is communicated to the node that wants to connect with our node.
+
+Once the .onion addresses have been created with the procedures oulined above, 
+to make your external IP address public you add: `--bind-addr=yourInternalAddress:port --announce-addr=yourexternalIPAddress:port`
+
+To make your external .onion addresses public you add: `--bind-addr=yourInternalIPAddress:port --announce-addr=yourexternalIPAddress:port --announce-addr=.onionAddressV2:port --announce-addr=.onionAddressV3:port`
+
+
+#### Case #7	c-lightning has no public IP address and a fixed TOR V3 service address and fixed TOR V2 service  address a 3rd non persisten V2 address
+
+External users can connect to this node by TOR V2 and V3 and a random V2 until next tor release, then also (V3 randomly).
+
+The Persistent addresses can be created with the steps outlined above.
+
+You are not obliged to announce the non persistent V2 address but if want to do it:
+
+`--addr=autotor:<torservice_ip:port>`
 
-to see your onion addr use
-cli/lightning-cli getinfo
+and also you must specify `--tor-service-password=yourpassword` (not the hash) to access the
 
-the .onion addr has not to be announced to other nodes
-but you can with --announce-addr=xxxxxxxxxxxxxxxxxxxxxxx.onion[:port]
+tor service at 9051 If you have protected them with the password (no additional options if
+they are protected with a cookie file. See above.).
 
-if the < port_global: 127.0.0.1:port_local > in torrc fit with your
-lightningd options
+To make your external .onion address (V2 and V3) public you add: `--bind-addr=yourInternalIPAddress:port --announce-addr=your.onionAddressV2:port --announce-addr=your.onionAddressV3:port`
 
---bind-addr=xxx.xxx.xxx.xxx:port_local
---addr=xxx.xxx.xxx.xxx:port_local
+NOTE: if you want both of them public you can repeat the --announce-addr option. If your node is NOT inside an internal network you can use `--addr=external` instead.
 
-you can use the cli command : connect peerid xxxxxxxxxxxxxxxx.onion port_global
 
-some examples:
+#### Case #8 	c-lightning has a public IP address and no TOR hidden service address,
 
-sudo lightningd/lightningd --network=testnet --bind-addr=127.0.0.1:1234
---proxy=127.0.0.1:9050 --addr=autotor:127.0.0.1:9051 (auto binds 9735 global <--> local 1234)
+The external address is communicated by the `--announce-addr=yourexternalIPAddress:port`
 
-this will try to generate an V2 auto hidden-service by reading the tor cookie file and
-also create an not announced local ip address at port 1234
-the .onion addr will show with the cli command getinfo
+but can connect to any V4/6 ip address via a IPV4/6 socks 5 proxy by specifing 
+`--proxy=127.0.0.1:9050 --always-use-proxy`.
 
-the node is accessible by connect peerid xxxxxxxxxxxxxxxx.onion 9735
-or local by connect peerID 127.0.0.1 1234
+References
```

Comment:
Should this be a `##` heading or something?

Reply:
Just a forgotten attempt: Markdown doesn't show the links below but text obviously does. 
So in markdown  the word "references" would appear alone. Better to drop it completely.

In the text version it may seem natural to see some link at the end without specifying.